conversant/prompt_chatbot.py

```diff
@@ -186,11 +188,103 @@ def _update_max_context_examples(
             "tokens"
         )
 
+    def should_stop(self, response:str)-> bool:
+        """Given a response, decides if it ends with a stop sequence and should 
+        be terminated."""
+        for stop_seq in self.client_config["stop_sequences"]: #make into function
+            if response.endswith(stop_seq):
+                return True
+        return False
+    
+    def get_max_reruns(self):
+        """Returns the maximum number of times partial_reply should be invoked."""
+        return int(self.client_config["max_tokens"]/TOKENS_PER_REQUEST)
+    
+    def partial_reply(self, query: str, is_from_scratch: bool):
+        """Replies to a query given a chat history.
+
+        The reply is then generated directly from a call to a LLM.
+
+        Used for all responses besides the initial one. 
+
+        Args:
+            query (str): A query passed to the prompt chatbot.
+            
+            is_from_scratch (bool): Tells the chatbot if the reply
+            
+            should be generated from scratch or it is the continuation
+            
+            of the previous chunk
+            
+        Returns:
+            Interaction: Dictionary of query and generated LLM response
+            
+            bool: Indicates if the generation should be stopped
+        """
+        current_prompt = self.get_current_prompt(query)
```

Comment:
this first part of the function seems similar to the beginning of `reply`, so you could create a separate function to wrap this if you think it makes sense

Reply:
makes sense thanks!